crates/sui-framework/build.rs

```diff
@@ -163,30 +164,53 @@ fn build_packages_with_move_config(
     // write out generated docs
     // TODO: remove docs of deleted files
     if write_docs {
-        for (fname, doc) in deepbook_pkg.package.compiled_docs.unwrap() {
+        let mut files_to_write = BTreeMap::new();
+        relocate_docs(
+            deepbook_dir,
+            &deepbook_pkg.package.compiled_docs.unwrap(),
+            &mut files_to_write,
+        );
+        relocate_docs(
+            system_dir,
+            &system_pkg.package.compiled_docs.unwrap(),
+            &mut files_to_write,
+        );
+        relocate_docs(
+            framework_dir,
+            &framework_pkg.package.compiled_docs.unwrap(),
+            &mut files_to_write,
+        );
+        for (fname, doc) in files_to_write {
             let mut dst_path = PathBuf::from(DOCS_DIR);
-            dst_path.push(deepbook_dir);
-            dst_path.push(fname);
-            fs::create_dir_all(dst_path.parent().unwrap()).unwrap();
-            fs::write(dst_path, doc).unwrap();
-        }
-        for (fname, doc) in system_pkg.package.compiled_docs.unwrap() {
-            let mut dst_path = PathBuf::from(DOCS_DIR);
-            dst_path.push(system_dir);
-            dst_path.push(fname);
-            fs::create_dir_all(dst_path.parent().unwrap()).unwrap();
-            fs::write(dst_path, doc).unwrap();
-        }
-        for (fname, doc) in framework_pkg.package.compiled_docs.unwrap() {
-            let mut dst_path = PathBuf::from(DOCS_DIR);
-            dst_path.push(framework_dir);
             dst_path.push(fname);
             fs::create_dir_all(dst_path.parent().unwrap()).unwrap();
             fs::write(dst_path, doc).unwrap();
         }
     }
 }
 
+fn relocate_docs(prefix: &str, files: &[(String, String)], output: &mut BTreeMap<String, String>) {
```

Comment:
nit: docs explaining the what and why of this function would be helpful.

Reply:
Also, this implementation is doing a depth-first traversal, IIUC, which means that currently the version of the `sui-framework` docs that get written will be the version that came from deepbook's dependencies. Is there value in ensuring that all the docs that get copied over are top-level docs, and then fixing up the links in those, or should it not matter?

Reply:
It shouldn't matter. The only difference between them will be the relative links to things outside of the package which we're already changing.